source/PlayerInfo.cpp

```diff
@@ -1346,6 +1346,17 @@ void PlayerInfo::ReorderShip(int fromIndex, int toIndex)
 	shared_ptr<Ship> ship = ships[fromIndex];
 	ships.erase(ships.begin() + fromIndex);
 	ships.insert(ships.begin() + toIndex, ship);
+	// Check if the ship in the first position can be a flagship and is in the current system.
+	vector<shared_ptr<Ship>>::iterator firstShipIt = ships.begin();
+	if(Preferences::Has("Automatically unpark flagship")
+			&& (*firstShipIt)->CanBeFlagship() && (*firstShipIt)->GetSystem() == system)
+	{
+		if(fromIndex == 0)
+			ships[toIndex]->SetIsParked(true);
+		else
+			ships[fromIndex]->SetIsParked(true);
+		(*firstShipIt)->SetIsParked(false);
```

Comment:
```suggestion
		firstShipIt->SetIsParked(false);
```

Reply:
It does not work without (*). Apperantly because this is an iterator and not the object itself. And the iterator has no connection to the functions

Reply:
Ah sorry forgot it was pointer to pointer, ->* should work too, but fine as is

Reply:
I thougt aswell that *firstShipIt-> should do it. But it does not. Somehow i really need the brackets